packages/docs-browser/src/components/Action/actions/Move.js

```diff
@@ -0,0 +1,106 @@
+import styled from 'styled-components'
+import _isEqual from 'lodash/isEqual'
+
+import DocsBrowser from '../../../main'
+import getNode from '../../../utils/getNode'
+import {initialize, moveElements} from '../../../modules/move/actions'
+
+const StyledButtonsWrapper = styled.div`
+  display: flex;
+  justify-content: flex-end;
+
+  ${StyledButton} {
+    margin-right: 0;
+  }
+`
+
+const MoveAction = ({selection, onSuccess, onError, context, initialize, moveElements, isWaiting}) => {
+  const initialNode = getNode(context.history.location.pathname)
+  const initialParent = {model: 'Docs_list_item', key: `${initialNode.model}/${initialNode.key}`}
+  const [parent, setParent] = useState(initialParent)
+
+  useEffect(() => {
+    initialize(selection, onSuccess, onError)
+  }, [])
+
+  const getCustomLocation = (model, key) => {
+    switch (model) {
+      case 'Domain':
+        return `/docs/domain/${key}/list`
+      case 'Folder':
+        return `/docs/folder/${key}/list`
+      case 'Resource':
+        return
+      default:
+        throw new Error(`Unexpected model: ${model}`)
+    }
+  }
+
+  const getInitialLocation = () => {
+    if (parent === null) {
+      return null
+    }
+    const [model, key] = parent.key.split('/')
+    return getCustomLocation(model, key)
+  }
+
+  return <LoadMask required={[!isWaiting]}>
+    <DocsBrowser
+      memoryHistory={true}
+      initialLocation={getInitialLocation()}
+      listLimit={10}
+      selectionStyle="none"
+      searchFormType="none"
+      disableViewPersistor={true}
+      listFormName={parent === null ? 'Move_root_docs_list_item' : 'Move_docs_list_item'}
+      onListParentChange={parent => setParent(parent)}
+      getCustomLocation={getCustomLocation}
+      navigationStrategy={{}}
+    />
+    <StyledButtonsWrapper>
+      <Button
+        onClick={() => moveElements(parent.key, selection.ids)}
+        look="raised"
+        ink="primary"
+        disabled={parent === null || _isEqual(parent, initialParent)}
+      >
+        <FormattedMessage id="client.actions.dms-move.button"/>
+      </Button>
+    </StyledButtonsWrapper>
+  </LoadMask>
+}
+
+MoveAction.propTypes = {
+  selection: selection.propType,
+  onSuccess: PropTypes.func.isRequired,
+  onError: PropTypes.func.isRequired,
+  context: PropTypes.shape({
+    history: PropTypes.shape({
+      location: PropTypes.shape({
+        pathname: PropTypes.string.isRequired
+      }).isRequired
+    }).isRequired
+  }).isRequired,
+  initialize: PropTypes.func.isRequired,
+  moveElements: PropTypes.func.isRequired,
+  isWaiting: PropTypes.bool.isRequired
+}
+
+const mapActionCreators = {
+  initialize,
+  moveElements
+}
+
+const mapStateToProps = state => ({
+  isWaiting: state.docs.move.isWaiting
+})
+
+const MoveActionContainer = connect(mapStateToProps, mapActionCreators)(injectIntl(MoveAction))
+
+export default injectIntl(props => <MoveActionContainer {...props} />)
```

Comment:
to be a bit consistent to other code i would create the MoveActionContainer in a separate file (same folder) and avoid two injectIntls.